src/main/java/org/wildfly/security/http/util/sso/DefaultSingleSignOn.java

```diff
@@ -61,7 +61,7 @@ public String getName() {
     }
 
     @Override
-    public SecurityIdentity getIdentity() {
+    public synchronized SecurityIdentity getIdentity() {
```

Comment:
:-1: There's no reason to synchronize this method.  If the synchronized modifier in the method signature of DefaultSingleSignOn.setIdentity(...) is cumbersome for subclasses, then there are better ways to address this (e.g. by replacing the modifier with a synchronized block, and clarifying the javadoc).